src/components/Metrics.js

```diff
@@ -15,7 +22,25 @@ const Metrics = React.memo(function Metrics({
 }) {
   const { layoutName } = useLayout()
   const compactMode = layoutName === 'small'
-  const { requestToken, stakeToken } = useAppState()
+  const { requestToken, stakeToken, currencies } = useAppState()
+  const [currencyIndex, setCurrencyIndex] = useState(0)
+  const currency = useMemo(() => {
+    if (!currencies.length) {
+      return {
+        name: 'USD',
+        symbol: '$',
+        rate: 1,
+      }
+    }
+
+    return currencies[currencyIndex]
+  }, [currencyIndex, currencies])
+  const handleCurrencyChange = useCallback(currencyIndex => {
+    setCurrencyIndex(currencyIndex)
+  }, [])
```

Comment:
This might not be needed. We can pass `setCurrencyIndex` directly to the dropdown `onChange` prop.